okapi-core/src/main/java/org/folio/okapi/util/DepResolution.java

```diff
@@ -736,7 +661,58 @@ static void installMaxIterations(
     if (!errors.isEmpty()) {
       throw new OkapiError(ErrorType.USER, String.join(". ", errors));
     }
-    sortTenantModules(tml, modsAvailable, modsEnabled);
+    sortTenantModules(tml, modsAvailable, enabledModules);
+
+  }
+
+  static void sortTenantModules(List<TenantModuleDescriptor> tml,
+      Map<String, ModuleDescriptor> modsAvailable, Collection<ModuleDescriptor> modules) {
+
+    logger.info("sortTenantModules with list {}",
+        tml.stream().map(TenantModuleDescriptor::getId).collect(Collectors.joining(", ")));
+    List<TenantModuleDescriptor> result = new ArrayList<>();
+    Iterator<TenantModuleDescriptor> iterator = tml.iterator();
+    while (iterator.hasNext()) {
+      TenantModuleDescriptor tm = iterator.next();
+      ModuleDescriptor md = modsAvailable.get(tm.getId());
+      if (tm.getAction().equals(TenantModuleDescriptor.Action.disable)) {
+        logger.debug("See if module {} can be removed from existing list of modules {}",
+            md.getId(), modules.stream().map(ModuleDescriptor::getId)
+                .collect(Collectors.joining(", ")));
+        if (DepResolution.moduleDepRequired(modules, md)) {
+          logger.debug("yes: removing {}", md.getId());
+          iterator.remove();
+          iterator = tml.iterator();
+          result.add(tm);
+          modules.remove(md);
+        }
+      } else if (tm.getAction().equals(TenantModuleDescriptor.Action.enable)) {
+        logger.debug("See if module {} can be added to existing list of modules {}",
+            md.getId(), modules.stream().map(ModuleDescriptor::getId)
+                .collect(Collectors.joining(", ")));
+        if (DepResolution.moduleDepProvided(modules, md)) {
+          logger.debug("yes: adding {}", md.getId());
+          iterator.remove();
+          iterator = tml.iterator();
+          result.add(tm);
+          modules.add(md);
+          String moduleFrom = tm.getFrom();
+          if (moduleFrom != null) {
+            logger.debug("yes: removing from {}", moduleFrom);
+            modules.remove(modsAvailable.get(moduleFrom));
+          }
+        }
+      } else {
+        iterator.remove();
+        iterator = tml.iterator();
+        result.add(tm);
+      }
+    }
+    if (!tml.isEmpty()) {
+      throw new OkapiError(ErrorType.USER, "Some modules cannot be topological sorted: "
```

Comment:
These modules cannot be topological sorted because they have circular or unsatisfiable interface dependencies:

Reply:
Exactly!

Reply:
Can you add the full explanation to the error message?